### ISSUE TITLE
`/* @once */` seems to work only for first child of object

### ISSUE THREAD

**Issue by titoBouzout** [ryansolid/dom-expressions#252]
Say we have 

```js
function Comp(props) {
  return (
    <div style={/* @once */ { width: props.width, height: props.height }} />
  );
}
```

generated code is

```js
function Comp(props) {
  return (() => {
    const _el$ = _tmpl$();

    props.width != null ? 
        _el$.style.setProperty("width", props.width) :
         _el$.style.removeProperty("width");

    _$effect(() => props.height != null ? 
        _el$.style.setProperty("height", props.height) :
         _el$.style.removeProperty("height"));

    return _el$;
  })();
}
```

Playground: https://playground.solidjs.com/anonymous/03735897-b261-4420-835c-7e69c27ad978

**Comment by ryansolid** [comment]
Yeah style and classList are special in that they analyze on a per object expression level. So in this case put the once in front of each property.

```js
<div style={ { width: /* @once */props.width, height: /* @once */props.height }} />
```

I realize there is nothing stopping us from oncing the whole object as well.

**Comment by edemaine** [comment]
The current behavior is particularly confusing when `style` is set to something other than an object literal:

```js
<div style={/* @once */ props.style} />
```

Currently `/* @once */` is ignored in this case.  [Playground](https://playground.solidjs.com/anonymous/70270d9f-0a25-4821-9dae-7b062a8d7cae)

In any case, I've [added some documentation](https://github.com/solidjs/solid-docs/commit/9ddbb157510375194374e4f3b96c95974f5488da) about the current behavior so this is less surprising.

**Comment by titoBouzout** [comment]
I was just looking how to fix this, the logic seems to be here https://github.com/ryansolid/dom-expressions/blob/main/packages/babel-plugin-jsx-dom-expressions/src/shared/utils.js#L87-L94 
Im staring at this not knowing what to do with that information

**Comment by titoBouzout** [comment]
https://github.com/ryansolid/dom-expressions/pull/258

**Comment by titoBouzout** [comment]
I heard the expectation is to use a memo instead, is the other approach?  Looks to me a bit of a waste, as there's going to be an effect created that is not going to be used, code would be bigger.

For the case in place, can you consider merging it regardless of the future of this feature, it makes it work closer to the intention and of what's expected.

**Comment by titoBouzout** [comment]
see also https://github.com/solidjs/solid/issues/1938
